elide-core/src/test/java/com/yahoo/elide/core/LifeCycleTest.java

```diff
@@ -395,6 +428,130 @@ public void testUpdateWithChangeSpec() {
+        verify(book, never()).onDeleteBook(scope);
+        verify(book, times(1)).onUpdateGenre(any(RequestScope.class), any(ChangeSpec.class));
+        verify(book, never()).onUpdateGenre(any(RequestScope.class), isNull());
+        verify(book, times(1)).preRead(scope);
+        verify(book, never()).alwaysOnUpdate();
+        verify(onUpdateDeferredCallback, never()).execute(eq(book), isA(RequestScope.class), any());
+        verify(onUpdateImmediateCallback, times(2)).execute(eq(book), isA(RequestScope.class), any());
+        verify(onUpdatePostCommitCallback, never()).execute(eq(book), isA(RequestScope.class), any());
+        verify(book, times(2)).checkPermission(scope);
+
+        scope.runQueuedPreSecurityTriggers();
+        verify(book, never()).onCreateBook(scope);
+        verify(book, never()).onDeleteBook(scope);
+        verify(book, times(1)).onUpdateGenre(any(RequestScope.class), any(ChangeSpec.class));
+        verify(book, times(1)).preRead(scope);
+        verify(book, never()).alwaysOnUpdate();
+        verify(onUpdateDeferredCallback, never()).execute(eq(book), isA(RequestScope.class), any());
+        verify(onUpdateImmediateCallback, times(2)).execute(eq(book), isA(RequestScope.class), any());
+        verify(onUpdatePostCommitCallback, never()).execute(eq(book), isA(RequestScope.class), any());
+        verify(book, times(2)).checkPermission(scope);
+
+        scope.runQueuedPreCommitTriggers();
+        verify(book, never()).preCreateBook(scope);
+        verify(book, never()).preDeleteBook(scope);
+        verify(book, times(1)).preUpdateGenre(any(RequestScope.class), any(ChangeSpec.class));
+        verify(book, times(1)).alwaysOnUpdate();
+        verify(onUpdateDeferredCallback, times(2)).execute(eq(book), isA(RequestScope.class), any());
+        verify(onUpdateImmediateCallback, times(2)).execute(eq(book), isA(RequestScope.class), any());
+        verify(onUpdatePostCommitCallback, never()).execute(eq(book), isA(RequestScope.class), any());
+        verify(book, times(2)).checkPermission(scope);
+
+        scope.getPermissionExecutor().executeCommitChecks();
+        verify(book, times(2)).checkPermission(scope);
+
+        scope.runQueuedPostCommitTriggers();
+        verify(book, never()).postCreateBook(scope);
+        verify(book, never()).postDeleteBook(scope);
+        verify(book, times(1)).postUpdateGenre(any(RequestScope.class), any(ChangeSpec.class));
+        verify(book, times(1)).postRead(scope);
+        verify(book, times(1)).alwaysOnUpdate();
+        verify(onUpdateDeferredCallback, times(2)).execute(eq(book), isA(RequestScope.class), any());
+        verify(onUpdateImmediateCallback, times(2)).execute(eq(book), isA(RequestScope.class), any());
+        verify(onUpdatePostCommitCallback, times(2)).execute(eq(book), isA(RequestScope.class), any());
+        verify(onUpdatePostCommitAuthor, never()).execute(any(), isA(RequestScope.class), any());
```

Comment:
This is technically redundant with the next call.

Reply:
Not exactly.  The next call checks only for empty Optional, but since it is called for all tests, we can leave it.
```java
verify(onUpdatePostCommitAuthor, never()).execute(any(), isA(RequestScope.class), eq(Optional.empty()));
```